docs/how-to-build-a-lexical-model.md

```diff
@@ -0,0 +1,365 @@
+
+
+Step 4.2: Creating the KPS file
+-------------------------------
+
+**TODO:**
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<Package>
+  <System>
+    <KeymanDeveloperVersion>12.0.1500.0</KeymanDeveloperVersion>
+    <FileVersion>12.0</FileVersion>
+  </System>
+  <Options>
+    <FollowKeyboardVersion/>
+  </Options>
+  <Info>
+    <Name URL="">SENĆOŦEN (Saanich Dialect) Lexical Model</Name>
+    <Copyright URL="">© 2019 National Research Council Canada</Copyright>
+    <Author URL="mailto:Eddie.Santos@nrc-cnrc.gc.ca">Eddie Antonio Santos</Author>
+    <Version>1.0.3</Version>
+  </Info>
+  <Files>
+    <File>
+      <Name>..\build\nrc.str.sencoten.model.js</Name>
+      <Description>Lexical model nrc.str.sencoten.model.js</Description>
+      <CopyLocation>0</CopyLocation>
+      <FileType>.model.js</FileType>
+    </File>
+  </Files>
+  <LexicalModels>
+    <LexicalModel>
+      <Name>SENĆOŦEN dictionary</Name>
+      <ID>nrc.str.sencoten</ID>
+      <Version>1.0.3</Version>
+      <Languages>
+        <Language ID="str">North Straits Salish</Language>
+        <Language ID="str-Latn">SENĆOŦEN</Language>
+      </Languages>
+    </LexicalModel>
+  </LexicalModels>
+</Package>
+```
+
+Step 4.3: Creating the model file
+---------------------------------
+
+Finally, the `model.ts` is a TypeScript source code file that is
+executed to build the model. For our purposes, we can simply copy-paste
+the following example:
+
+```typescript
+import LexicalModelCompiler from "@keymanapp/developer-lexical-model-compiler";
+
+(new LexicalModelCompiler).compile({
+  format: 'trie-1.0',
+  wordBreaking: 'default',
+  sources: ['wordlist.tsv'],
+});
+```
+
+Most importantly, this file specifies that model will use the word list
+in `wordlist.tsv`.
+
+For simple lexical models with exactly one wordlist called
+`wordlist.tsv`, **this file does not need to be changed any further**.
+
+However, it specifies a few other things:
+
+  - `format: 'trie-1.0'`: Use the trie back-end, that allows for
+    prediction based on prefixes of words.
+  - `wordBreaking: 'default'`: use Unicode's default word boundary rules
+    to find where "words" are when predicting text. This works for most
+    languages and writing systems, however, it does not work for
+    languages/writing systems that do not mark breaks between words,
+    e.g., Chinese, Thai, Laos, Khmer, and Japanese, to name a few. For
+    these writing systems, you must provide a language-specific word
+    breaking algorithm to find the words in running text.
+
+Step 5: Build the model
+-----------------------
+
+
+Appendix: Tab-separated wordlist file format specification
+----------------------------------------------------------
+
+<a id="appendix:tsv"></a>
+
+[appendix]: ./#appendix:tsv
+
+The **lexical model compiler** expects wordlists to abide by the
+following **tab-separated values** (TSV) format:
+
+ - the file is must be **UTF-8** encoded text file
```

Comment:
```suggestion
 - the file must be **UTF-8** encoded text file without Byte Order Mark (BOM).
```

Note: Notepad until recently always prepended a BOM. This is now changing: https://www.bleepingcomputer.com/news/microsoft/windows-10-notepad-is-getting-better-utf-8-encoding-support/

Reply:
I mean, is it hard to strip out the BOM?  Seems like a big ask for those not so familiar with byte encoding particularities.

Reply:
Good point! It's not really that difficult, and I don't _think_ there's any problem here. I don't think... please debate it over here: keymanapp/keyman#1916!